AsyncDisplayKit/ASMapNode.mm

```diff
@@ -207,7 +207,10 @@ - (void)takeSnapshot
                     [image drawAtPoint:CGPointZero];
                     
                     // Get a standard annotation view pin. Future implementations should use a custom annotation image property.
-                    MKAnnotationView *pin = [[MKPinAnnotationView alloc] initWithAnnotation:nil reuseIdentifier:@""];
+                    __block MKAnnotationView *pin;
+                    dispatch_sync(dispatch_get_main_queue(), ^{
```

Comment:
Unfortunately, dispatch_sync to the main thread is not allowed by the AsyncDisplayKit threading model, and can result in deadlocks if this node happens to be part of a hierarchy that undergoes synchronous layout or display.  The threading model allows for blocking the main thread on background threads (clearly this is to be avoided when possible, but is sometimes necessary for visual correctness) — but a consequence is that the reverse is never allowed.

Since -takeSnapshot is not itself synchronous, we should be able to (if necessary) dispatch_async to main and then back again.

If we made use of this image in all maps, I'd say we should capture it in +initialize, but since we don't always use this let's avoid that.